src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_InvalidExpression.cs

```diff
@@ -0,0 +1,390 @@
+      Right: IPropertyReferenceExpression: System.Int32 System.Array.Length { get; } (OperationKind.PropertyReferenceExpression, Type: System.Int32)
+          Instance Receiver: IParameterReferenceExpression: args (OperationKind.ParameterReferenceExpression, Type: System.String[])
+";
+            VerifyOperationTreeForTest<BinaryExpressionSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/18057"), WorkItem(17598, "https://github.com/dotnet/roslyn/issues/17598")]
+        public void InvalidLambdaBinding_UnboundLambda()
+        {
+            string source = @"
+using System;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        var x /*<bind>*/= () => F()/*</bind>*/;
+    }
+
+    static void F()
+    {
+    }
+}
+";
+
+            string expectedLambdaOperationTree = @"
+ILambdaExpression (Signature: lambda expression) (OperationKind.LambdaExpression, Type: null)
+  IBlockStatement (1 statements) (OperationKind.BlockStatement)
+    IExpressionStatement (OperationKind.ExpressionStatement)
+      IInvocationExpression (static void Program.F()) (OperationKind.InvocationExpression, Type: System.Void)
+";
+            VerifyOperationTreeForTest<ParenthesizedLambdaExpressionSyntax>(source, expectedLambdaOperationTree);
+
+            string expectedEqualsValueOperationTree = @"
+IVariableDeclarationStatement (1 variables) (OperationKind.VariableDeclarationStatement, IsInvalid)
+  IVariableDeclaration: var x (OperationKind.VariableDeclaration)
+    Initializer: IUnboundLambdaExpression (OperationKind.UnboundLambdaExpression, Type: null)
+        ILambdaExpression (Signature: lambda expression) (OperationKind.LambdaExpression, Type: null)
+          IBlockStatement (1 statements) (OperationKind.BlockStatement)
+            IExpressionStatement (OperationKind.ExpressionStatement)
+              IInvocationExpression (static void Program.F()) (OperationKind.InvocationExpression, Type: System.Void)
+";
+            VerifyOperationTreeForTest<EqualsValueClauseSyntax>(source, expectedEqualsValueOperationTree);
+        }
+
+        [Fact, WorkItem(17598, "https://github.com/dotnet/roslyn/issues/17598")]
+        public void InvalidFieldInitializer()
+        {
+            string source = @"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+class Program
+{
+    int x /*<bind>*/= Program/*</bind>*/;
+    static void Main(string[] args)
+    {
+        var x = new Program() { x = Program };
+    }
+}
+";
+            string expectedOperationTree = @"
+IFieldInitializer (Field: System.Int32 Program.x) (OperationKind.FieldInitializerAtDeclaration, IsInvalid)
+  IConversionExpression (ConversionKind.Invalid, Implicit) (OperationKind.ConversionExpression, Type: System.Int32, IsInvalid)
+    IInvalidExpression (OperationKind.InvalidExpression, Type: Program, IsInvalid)
+      IOperation:  (OperationKind.None)
```

Comment:
I think it might be more beneficial to dump the type name of the Bound node for cases where it implements the base IOperation. @dotnet/analyzer-ioperation do you agree?